istioctl/cmd/validation.go

```diff
@@ -0,0 +1,414 @@
+	flags.StringVar(&opts.validatingWebhookServiceName, "validating-webhook-service-name", "istio-galley",
+		"The service name of the validating webhook to manage.")
+	flags.IntVar(&opts.validatingWebhookServicePort, "validating-webhook-service-port", 443,
+		"The service port of the validating webhook to manage.")
+	flags.StringVar(&opts.validatingWebhookNamespace, "validating-webhook-namespace", "istio-system",
+		"The namespace of the validating webhook to manage.")
+	flags.IntVar(&opts.maxTimeForCheckingWebhookServer, "max-time-for-checking-webhook-server", 60,
+		"	Max time (in seconds) for checking the validating webhook server. If the validating webhook server is not ready"+
+			"in the given time, exit. Otherwise, apply the webhook configuration.")
+
+	return validationCmd
+}
+
+// Create the validatingwebhookconfiguration
+func createValidatingWebhookConfig(k8sClient kubernetes.Interface,
+	config *v1beta1.ValidatingWebhookConfiguration) (*v1beta1.ValidatingWebhookConfiguration, error) {
+	if config == nil {
+		return nil, fmt.Errorf("validatingwebhookconfiguration is nil")
+	}
+	client := k8sClient.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations()
+	_, err := client.Get(config.Name, metav1.GetOptions{})
+	if err == nil {
+		// Delete the existing webhook configuration
+		if delErr := client.Delete(config.Name, &metav1.DeleteOptions{}); delErr != nil {
+			fmt.Printf("when deleting webhook configuration %v, return: %v\n", config.Name, delErr)
+		}
+		time.Sleep(5 * time.Second)
+	}
+	fmt.Printf("create webhook configuration %v\n", config.Name)
+	whConfig, err := client.Create(config)
+	return whConfig, err
+}
+
+// Read CA certificate and check whether it is a valid certificate.
+func readCACert(client kubernetes.Interface, certPath string, readLocal bool) ([]byte, error) {
+	var caCert []byte
+	var err error
+	if readLocal {
+		caCert, err = ioutil.ReadFile(certPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read CA cert, cert. path: %v, error: %v", certPath, err)
+		}
+	} else {
+		// Create a pod and wait it is ready before reading k8s CA certificate.
+		// If the pod already exists, proceed to reading k8s CA certificate from the pod.
+		if err = createPod(client, nameForTheTempPod, namespaceForTheTempPod, labelKeyForTheTempPod,
+			labelValForTheTempPod, containerNameForTheTempPod, containerImageForTheTempPod,
+			containerCmdForTheTempPod); err != nil && !errors.IsAlreadyExists(err) {
+			return nil, fmt.Errorf("error when creating pod and err is not already exists: %v", err)
+		}
+		fmt.Printf("pod %v created\n", nameForTheTempPod)
+		fmt.Printf("waiting for at most %v until the pod is running ...\n", maxPodWaitDuration)
+		err = waitForPodRunning(client, nameForTheTempPod, labelKeyForTheTempPod, labelValForTheTempPod,
+			namespaceForTheTempPod, maxPodWaitDuration)
+		if err != nil {
+			return nil, fmt.Errorf("error when waiting for the pod: %v", err)
+		}
+		execCmd := "cat " + certPath
```

Comment:
Is there are another way to do this? we cannot exec to distroless

Reply:
Done: changed to another way to do it.